php/read_file.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/* Read a file (file path given as a command line argument),
+and write to stdout
+*/
+
+// Get the file path from command line arguments
+$file_path = $argv[1];
+
+// Open the file for reading
+$file = fopen($file_path, 'r');
+
+$i = 1;
+
+while (($line = fgets($file)) !== false) {
+    echo $i . ' ' . strtoupper($line);
+    $i++;
+}
```

Comment:
I think we could replace most of this with:

```php
foreach (file($file_path) as $index => $line) {
    echo ($index + 1) . ' ' . strtoupper($line);
}
```

The [`file` function](https://www.php.net/manual/en/function.file.php) reads a file and returns its lines in an array.

Reply:
Cool! Looks like `file()` also closes the file after reading it, so that makes this script v succinct